server/socket.js

```diff
@@ -151,4 +151,80 @@ module.exports = function(io) {
     return ready;
   }
 
+  function updatePlayerStatsInDatabase() {
+    // Update game status
+    var gamesref = dataBase.ref(`games/` + gameId.key);
+    var winner = getAllPlayers();
+    gamesref.update({winner: winner[0].id, status: "finished"});
+
+    var allPlayers = getAllPlayersAliveOrDead();
+    
+    // If it's a 1v1, we update the ratings of each player, else don't update ratings
+    var updateRatings = allPlayers.length === 2;
+    if (updateRatings) {
+
+      var winner = getAllPlayers()[0];  // winner is the only one alive
+      var winnerData = {};
+      for (var i = 0; i < dbPlayers.length; i++) {
+        if (dbPlayers[i].displayName === winner.id) {
+          winnerData = dbPlayers[i];
+        }
+      }
+
+      var loserData = {};
+      for (var i = 0; i < dbPlayers.length; i++) {
+        if (dbPlayers[i].displayName !== winner.id) {
+          loserData = dbPlayers[i];
+        }
+      }
+
+      var winnerNewRating = eloCalc.getRatingIfWin(winnerData.rating, loserData.rating);
+      var loserNewRating = eloCalc.getRatingIfLose(loserData.rating, winnerData.rating);
+    }       
+
+    var gamesRef = dataBase.ref(`games/` + gameId.key);
+    var usersRef = dataBase.ref(`users/`);
+
+    // Update winner's wins, pearls, rating
+    var query = usersRef.orderByChild("displayName").equalTo(winner.id);
+    query.once("value", function(snapshot) {
+      var users = snapshot.val();
+      var userKeys = Object.keys(users);
+      var winnerKey = userKeys[0];
+      var winner = users[winnerKey];
+
+      var winnerRef = dataBase.ref(`users/` + winnerKey);
+      if (updateRatings) {
+        winnerRef.update({wins: winner.wins + 1, pearls: winner.pearls + PEARLS_ON_WIN, rating: winnerNewRating});
+      } else {
+        winnerRef.update({wins: winner.wins + 1, pearls: winner.pearls + PEARLS_ON_WIN});
+      }
+    });
+
+    // Update all losers' losses, pearls, rating
+    var allPlayers = getAllPlayersAliveOrDead();
+    for (var i = 0; i < allPlayers.length; i++) {
+      
+      // Update all losers in database
+      if (allPlayers[i].id !== winner.id) {
+        var id = allPlayers[i].id;
+
+        var query = usersRef.orderByChild("displayName").equalTo(id);
+        query.once("value", function(snapshot) {
+          var users = snapshot.val();
+          var userKeys = Object.keys(users);
+          var userKey = userKeys[0];
+          var user = users[userKey];
+
+          var loserRef = dataBase.ref(`users/` + userKey);
+
+          if (updateRatings) {
+            loserRef.update({losses: user.losses + 1, pearls: user.pearls + PEARLS_ON_LOSE, rating: loserNewRating});
+          } else {
+            loserRef.update({losses: user.losses + 1, pearls: user.pearls + PEARLS_ON_LOSE});
+          }
+        });
+      }
+    }
+  }
```

Comment:
looks good, maybe in the future try breaking it out into several functions to try to keep it a little bit more modular and less spaghetti-ish.